### ISSUE TITLE
[BUG]: AUC discrepancy with sklearn

### ISSUE THREAD

**Issue by josp70** [pycaret/pycaret#3686]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I'm trying to reproduce the AUC metric using sklearn.metrics.roc_auc_score. The reported AUC from predict_model is 0.8077 while the computed with sklearn is "0.3016".

```
pycaret version is 3.0.4
sklearn version is 1.2.2
```

### Reproducible Example

```python
from pycaret.datasets import get_data
from pycaret.classification import ClassificationExperiment
from sklearn.metrics import roc_auc_score
from pycaret import __version__ as pycaret_version
from sklearn import __version__ as sklearn_version

print(f"pycaret version is {pycaret_version}")
print(f"sklearn version is {sklearn_version}")

data = get_data('diabetes', verbose=False)

exp = ClassificationExperiment()
target = 'Class variable'
exp.setup(data, target=target, session_id=123)

model = exp.create_model("xgboost")

pred = exp.predict_model(model, probability_threshold=0.5)

print(pred[[target, 'prediction_label',  'prediction_score']])

metrics_df = exp.pull()

metrics_pycaret = {'AUC': metrics_df.AUC.values[0]}
print(metrics_pycaret)

metrics_sklearn = {'AUC': roc_auc_score(pred[target].values, pred.prediction_score.values)}
print(metrics_sklearn)
```

### Expected Behavior

I expect `metrics_sklearn.AUC == metrics_pycaret.AUC`

### Actual Results

```python-traceback
pycaret version is 3.0.4
sklearn version is 1.2.2
                    Description             Value
0                    Session id               123
1                        Target    Class variable
2                   Target type            Binary
3           Original data shape          (768, 9)
4        Transformed data shape          (768, 9)
5   Transformed train set shape          (537, 9)
6    Transformed test set shape          (231, 9)
7              Numeric features                 8
8                    Preprocess              True
9               Imputation type            simple
10           Numeric imputation              mean
11       Categorical imputation              mode
12               Fold Generator   StratifiedKFold
13                  Fold Number                10
14                     CPU Jobs                -1
15                      Use GPU             False
16               Log Experiment             False
17              Experiment Name  clf-default-name
18                          USI              70f3
      Accuracy     AUC  Recall   Prec.      F1   Kappa     MCC
Fold
0       0.6667  0.7955  0.4737  0.5294  0.5000  0.2512  0.2520
1       0.7037  0.7519  0.6316  0.5714  0.6000  0.3656  0.3668
2       0.7407  0.8195  0.4737  0.6923  0.5625  0.3874  0.4014
3       0.6667  0.8045  0.4737  0.5294  0.5000  0.2512  0.2520
4       0.7222  0.8286  0.5789  0.6111  0.5946  0.3836  0.3839
5       0.6296  0.6466  0.4737  0.4737  0.4737  0.1880  0.1880
6       0.6667  0.7203  0.4737  0.5294  0.5000  0.2512  0.2520
7       0.8302  0.8397  0.6111  0.8462  0.7097  0.5940  0.6098
8       0.6226  0.6492  0.3889  0.4375  0.4118  0.1354  0.1359
9       0.6038  0.6603  0.3333  0.4000  0.3636  0.0794  0.0801
Mean    0.6853  0.7516  0.4912  0.5620  0.5216  0.2887  0.2922
Std     0.0636  0.0734  0.0887  0.1237  0.0945  0.1413  0.1456
                       Model  Accuracy     AUC  Recall  Prec.      F1   Kappa     MCC
0  Extreme Gradient Boosting    0.7359  0.8077  0.6173  0.625  0.6211  0.4185  0.4185
     Class variable  prediction_label  prediction_score
552               0                 0            0.9898
438               0                 0            0.9999
149               0                 0            0.9998
373               0                 0            0.9573
36                0                 1            0.5514
..              ...               ...               ...
85                0                 1            0.6434
7                 0                 1            0.8367
298               1                 0            0.7122
341               0                 0            0.9986
472               0                 1            0.6539

[231 rows x 3 columns]
{'AUC': 0.8077}
{'AUC': 0.30156378600823047}
```

### Installed Versions

<details>
System:
    python: 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
executable: D:\Dev\source\data-science\luigi-poc\.venv\Scripts\python.exe
   machine: Windows-10-10.0.19044-SP0

PyCaret required dependencies:
                 pip: 23.1.2
          setuptools: 68.0.0
             pycaret: 3.0.4
             IPython: 7.34.0
          ipywidgets: 7.7.5
                tqdm: 4.65.0
               numpy: 1.23.0
              pandas: 1.5.3
              jinja2: 3.1.2
               scipy: 1.10.1
              joblib: 1.3.1
             sklearn: 1.2.2
                pyod: 1.1.0
            imblearn: 0.11.0
   category_encoders: 2.6.1
            lightgbm: 4.0.0
               numba: 0.57.1
            requests: 2.31.0
          matplotlib: 3.7.2
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.15.0
    plotly-resampler: Not installed
             kaleido: 0.2.1
           schemdraw: 0.15
         statsmodels: 0.14.0
              sktime: 0.20.1
               tbats: 1.1.3
            pmdarima: 2.0.3
              psutil: 5.9.5
          markupsafe: 2.1.3
             pickle5: Not installed
         cloudpickle: 2.2.1
         deprecation: 2.1.0
              xxhash: 3.2.0
           wurlitzer: Not installed

PyCaret optional dependencies:
                shap: 0.42.1
           interpret: 0.4.2
                umap: 0.5.3
    pandas_profiling: 4.3.1
  explainerdashboard: 0.4.2.2
             autoviz: 0.1.730
           fairlearn: 0.7.0
          deepchecks: 0.17.3
             xgboost: 1.7.6
            catboost: 1.2
              kmodes: 0.12.2
             mlxtend: 0.22.0
       statsforecast: 1.5.0
        tune_sklearn: 0.4.6
                 ray: 2.5.1
            hyperopt: 0.2.7
              optuna: 3.2.0
               skopt: 0.9.0
              mlflow: 1.30.1
              gradio: 3.37.0
             fastapi: 0.100.0
             uvicorn: 0.23.0
              m2cgen: 0.10.0
           evidently: 0.2.8
               fugue: 0.8.5
           streamlit: Not installed
             prophet: Not installed
</details>

**Comment by josp70** [comment]
If you look at the prediction table you can see that for instance some observation with high predicted probabilities has the prediction_label as 0, where they should be 1, see for instance the observation with index 552 was computed with prediction_score= 0.9898 and the prediction_label=0

**Comment by moezali1** [comment]
@josp70 There are 2 questions here.

1. The difference between sklearn auc and pycaret is probably due to the fact that average parameter is setup to `macro` in sklearn calculation whereas it is set to `average` in pycaret.

![image](https://github.com/pycaret/pycaret/assets/54699234/bef7d9bd-d50c-4f2d-82f9-5fb7a96718bd)

2. The `prediction_score` in pycaret is of `prediction_label`. In your example, the way you interpret this is there is 0.9898 probability of this index being 0.
